receiver/mongodbatlasreceiver/receiver.go

```diff
@@ -231,3 +243,105 @@ func (s *receiver) extractProcessDiskMetrics(
 	}
 	return nil
 }
+
+// Log receiver logic
+func (s *receiver) Start(ctx context.Context, host component.Host) error {
+	go func() {
+		s.KickoffReceiver(ctx)
+	}()
+	return nil
+}
+
+func (s *receiver) Shutdown(ctx context.Context) error {
+	for _, stopperChan := range s.stopperChanList {
+		close(stopperChan)
+	}
+	ctx.Done()
+	return nil
+}
+
+func (s *receiver) getHostLogs(groupID, hostname, logName string) ([]model.LogEntry, error) {
+	// Get gzip bytes buffer from API
+	buf, err := s.client.GetLogs(context.Background(), groupID, hostname, logName, collection_interval)
+	if err != nil {
+		return nil, err
+	}
+	// Pass this into a gzip reader for decoding
+	reader, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Logs are in JSON format so create a JSON decoder to process them
+	dec := json.NewDecoder(reader)
+
+	entries := make([]model.LogEntry, 0)
+	for {
+		var entry model.LogEntry
+		err := dec.Decode(&entry)
+		if errors.Is(err, io.EOF) {
+			return entries, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, entry)
+	}
+}
+
+func (s *receiver) getHostAuditLogs(groupID, hostname, logName string) ([]model.AuditLog, error) {
+	// Get gzip bytes buffer from API
+	buf, err := s.client.GetLogs(context.Background(), groupID, hostname, logName, collection_interval)
+	if err != nil {
+		return nil, err
+	}
+	// Pass this into a gzip reader for decoding
+	reader, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Logs are in JSON format so create a JSON decoder to process them
+	dec := json.NewDecoder(reader)
+
+	entries := make([]model.AuditLog, 0)
+	for {
+		var entry model.AuditLog
+		err := dec.Decode(&entry)
+		if errors.Is(err, io.EOF) {
+			return entries, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, entry)
+	}
+}
+
+func (s *receiver) sendLogs(r resourceInfo, logName string) {
+	logs, err := s.getHostLogs(r.Project.ID, r.Hostname, logName)
+	if err != nil && err != io.EOF {
+		s.log.Warn("Failed to retreive logs", zap.Error(err))
+	}
+
+	for _, log := range logs {
+		r.LogName = logName
+		plog := mongodbEventToLogData(s.log, &log, r)
+		s.consumer.ConsumeLogs(context.Background(), plog)
+	}
+}
+
+func (s *receiver) sendAuditLogs(r resourceInfo, logName string) {
+	logs, err := s.getHostAuditLogs(r.Project.ID, r.Hostname, logName)
+	if err != nil && err != io.EOF {
+		s.log.Warn("Failed to retreive logs", zap.Error(err))
+	}
+
+	for _, log := range logs {
+		r.LogName = logName
+		plog := mongodbAuditEventToLogData(s.log, &log, r)
+		s.consumer.ConsumeLogs(context.Background(), plog)
+	}
+}
```

Comment:
These functions should probably go into `logs.go` as they are only used there. They seem out of place in this file. Either that or move the log collection functions in `logs.go` to this file if they are methods on the `receiver` object.